### PR TITLE
Add Strict-Transport-Security (HSTS) support

### PR DESCRIPTION
**Pull request by ghost**
This adds the Strict-Transport-Security header with a default value of 1 year.  It is enabled by default and will consequently break web apps that do not use HTTPS.

### CODE REVIEW

**Comment by qinfchen** [comment]
@jmcampanini

**Comment by jmcampanini** [comment]
we originally had HSTS available but turned off by default in the bundle. the reasoning for removing it are:
- it doesn't sound right that a single dropwizard service owns a whole domain - instead it feels like it would be an SPP responsibility
- setting the HSTS header incorrectly comes at a high cost (very hard to undo)

thoughts? i'd be open to adding it in, but i'd want the default to be OFF if we did.

File: src/main/java/com/palantir/websecurity/filters/WebSecurityHeaderInjector.java [hunk 1]
```diff
@@ -49,6 +52,10 @@ public void injectHeaders(HttpServletRequest request, HttpServletResponse respon
         checkNotNull(request);
         checkNotNull(response);
 
+        if (!this.strictTransportSecurity.isEmpty()) {
+            response.setHeader(HttpHeaders.STRICT_TRANSPORT_SECURITY, this.strictTransportSecurity);
```
Comment (jmcampanini) [inline src/main/java/com/palantir/websecurity/filters/WebSecurityHeaderInjector.java:56]: shouldn't this only be added if the request is HTTPS?

**Comment by ghost** [comment]
Yep, makes sense that this header should live with the reverse proxy when you have multiple apps behind the same proxy instance.  In the case where you're not using a proxy it's not particularly onerous to spin up nginx and configure this header. I'll close out this PR.

**Comment by jmcampanini** [comment]
cool. would be willing to revisit if this comes up in the field more. maybe add it as an off-by-default option (with a big warning).

**Pull request closed without merging**